File: python/run-length-encoding/run_length_encoding.py
from itertools import groupby
from re import sub

def decode(txt):
    return sub(r'(\d+)(\D)', _duplicate, txt)

def encode(txt):
    colated = [_count(group)+key for key, group in groupby(txt)]
    return ''.join(colated)

def _count(group):
	n = len(list(group))
	return (str(n), '')[n == 1]

def _duplicate(m):
	return m.group(2) * int(m.group(1))

File: javascript/run-length-encoding/run-length-encoding.js

function encode(str) {
    return group(str)
        .map(chars => {
            if (chars.length > 1) {
                return String(chars.length) + chars[0];
            }
            return chars[0];
        })
        .join('');
}

function decode(str) {
    let multiplier = 1;
    return str
        .split(/(\d+)(\D)/)
        .reduce((acc, val) => {
            if (!val) {
                multiplier = 1;
                return acc;
            }
            if (isNaN(val)) {
                acc = acc + Array(multiplier).fill(val).join('');
                multiplier = 1;
                return acc;
            }
            multiplier = Number.parseInt(val, 10);
            return acc;
        }, '');

}

function group(str) {
    let last = '';
    return str
        .split('')
        .reduce((acc, val) => {
            if (last === val) {
                acc[acc.length - 1] += val;
            } else {
                acc.push(val);
            }
            last = val;
            return acc;
        }, []);
}

module.exports = {
    encode: encode,
    decode, decode
};